packages/analytics-core/src/plugins/destination.ts

```diff
@@ -80,14 +79,19 @@ export class Destination implements DestinationPlugin {
   }
 
   addToQueue(...list: Context[]) {
+    const dropEventSet = new Set<string>();
     const tryable = list.filter((context) => {
       if (context.attempts < this.config.flushMaxRetries) {
         context.attempts += 1;
+
         return true;
       }
+      //Remove the events more than max retries from the storage.
+      dropEventSet.add(context.event.event_type);
       void this.fulfillRequest([context], 500, MAX_RETRIES_EXCEEDED_MESSAGE);
       return false;
     });
+    void this.updateEvent(dropEventSet);
```

Comment:
CASE 1: when an event exceeds the maximum retry limit, it should be deleted from the storage.